activerecord/test/cases/connection_pool_test.rb

```diff
@@ -373,15 +380,15 @@ def test_checkout_fairness
         errors = []
 
         threads = expected.map do |i|
-          t = Thread.new {
+          t = new_thread {
             begin
               @pool.checkout # never checked back in
```

Comment:
This line hangs when testing fibers, because without implementing a Ruby 3.1 scheduler that can hook into the blocking pool mutex wait, the fiber doesn't yield execution to any other fibers and simply hits the connection pool timeout.